File: TodoList/app/src/main/java/com/github/fredrik9000/todolist/add_edit_todo/TimePickerFragment.java
package com.github.fredrik9000.todolist.add_edit_todo;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.content.Context;
import android.os.Bundle;
import android.text.format.DateFormat;
import android.util.Log;
import android.widget.TimePicker;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import java.util.Calendar;

public class TimePickerFragment extends DialogFragment
        implements TimePickerDialog.OnTimeSetListener {

    private static final String TAG = "TimePickerFragment";
    private OnSelectTimeDialogInteractionListener listener;

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        // Use the current time as the default values for the picker
        final Calendar c = Calendar.getInstance();
        int hour = c.get(Calendar.HOUR_OF_DAY);
        int minute = c.get(Calendar.MINUTE);

        // Create a new instance of TimePickerDialog and return it
        return new TimePickerDialog(getActivity(), this, hour, minute,
                DateFormat.is24HourFormat(getActivity()));
    }

    @Override
    public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
        listener.onSelectTimeDialogInteraction(hourOfDay, minute);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        try {
            listener = (OnSelectTimeDialogInteractionListener) getTargetFragment();
        } catch (ClassCastException e) {
            Log.e(TAG, "onAttach: ClassCastException" + e.getMessage());
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }

    public interface OnSelectTimeDialogInteractionListener {
        void onSelectTimeDialogInteraction(int hour, int minute);
    }
}

File: TodoList/app/src/main/java/com/github/fredrik9000/todolist/add_edit_todo/TimePickerFragment.kt
package com.github.fredrik9000.todolist.add_edit_todo

import android.app.Dialog
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import android.text.format.DateFormat
import android.widget.TimePicker
import androidx.core.os.bundleOf
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.setFragmentResult
import java.util.*

class TimePickerFragment : DialogFragment(), OnTimeSetListener {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        // Uses the current time as the default values for the picker
        with(Calendar.getInstance()) {
            return TimePickerDialog(activity, this@TimePickerFragment, this[Calendar.HOUR_OF_DAY], this[Calendar.MINUTE], DateFormat.is24HourFormat(activity))
        }
    }

    override fun onTimeSet(view: TimePicker?, hour: Int, minute: Int) {
        setFragmentResult(TIME_PICKER_FRAGMENT_REQUEST_KEY, bundleOf(BUNDLE_HOUR_KEY to hour, BUNDLE_MINUTE_KEY to minute))
    }

    companion object {
        const val TIME_PICKER_FRAGMENT_REQUEST_KEY = "TIME_PICKER_FRAGMENT_REQUEST"
        const val BUNDLE_HOUR_KEY = "hour"
        const val BUNDLE_MINUTE_KEY = "minute"
    }
}